packages/components/src/filters/advanced/number-filter.js

```diff
@@ -0,0 +1,184 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import { Component, Fragment } from '@wordpress/element';
+import { SelectControl, TextControl } from '@wordpress/components';
+import { get, find, partial } from 'lodash';
+import interpolateComponents from 'interpolate-components';
+import classnames from 'classnames';
+import { _x } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import TextControlWithAffixes from '../../text-control-with-affixes';
+import { formatCurrency } from '../../../../currency/src';
+
+class NumberFilter extends Component {
+	getBetweenString() {
+		return _x(
+			'{{moreThan /}} and {{lessThan /}}',
+			'Numerical range inputs arranged on a single line',
+			'wc-admin'
+		);
+	}
+
+	getLegend( filter, config ) {
+		const inputType = get( config, [ 'input', 'type' ], 'number' );
+		const rule = find( config.rules, { value: filter.rule } ) || {};
+		let filterStr = filter.value || '';
+
+		if ( 'between' === rule.value ) {
+			let [ moreThan, lessThan ] = filterStr.split( ',' );
+
+			if ( 'currency' === inputType ) {
+				moreThan = formatCurrency( moreThan );
+				lessThan = formatCurrency( lessThan );
+			}
+
+			filterStr = interpolateComponents( {
+				mixedString: this.getBetweenString(),
+				components: {
+					moreThan: <Fragment>{ moreThan }</Fragment>,
+					lessThan: <Fragment>{ lessThan }</Fragment>,
+				},
+			} );
+		} else if ( 'currency' === inputType ) { // need to format a single input value as currency
+			filterStr = formatCurrency( filterStr );
+		}
+
+		return interpolateComponents( {
+			mixedString: config.labels.title,
+			components: {
+				filter: <span>{ filterStr }</span>,
+				rule: <span>{ rule.label }</span>,
```

Comment:
When `filterStr` is empty, this leads to a strange screen reader label for the fieldset– "Item Quantity is Between and", for example. Can you add a new label for the legend so that it reads "Item quantity is between two numbers" or even just always use "Item quantity limits"?

Reply:
So I actually see that we're doing this in all the filters, automatically updating the legend with the completed filter, which means all of them sound strange when the filter is empty.

I think the best solution would be to update all filters (in another PR), so that each legend is just the title, and there can be a hidden screen reader string that displays the "built" sentence– so the output HTML would look something like this:

```html
<fieldset>
	<legend>Item Quantity</legend>
	<div className="woocommerce-filters-advanced__fieldset">
		<span aria-hidden="true">Item Quantity is</span>
		<select aria-label="Item quantity filter match" />
		<input aria-label="Item quantity minimum amount" />
		<span aria-hidden="true">and</span>
		<input aria-label="Item quantity maximum amount" />
	</div>
	<span class="screen reader text">Item quantity is between 0 and 10</span>
</fieldset>
```

And that last span would only exist if the filter values are set.

Reply:
Opened an in-progress PR to solicit feedback on my approach: https://github.com/woocommerce/wc-admin/pull/1099

I just modified the `<legend>` because the `<span>` in addition to the `<legend>` sounded confusing in testing (though I'm still getting used to how VoiceOver reads things).